### ISSUE TITLE
Wrangler S3 read_csv API is not honouring "encoding_errors=ignore" option

### ISSUE THREAD

**Issue by Sahana-Bhat-GOQ** [aws/aws-sdk-pandas#1668]
### Describe the bug

I have a file in a csv format with data encoded other than `utf-8`. I want to ignore the malformed data and retain only the characters which are `utf-8` encoded. Pandas `read_csv` supports an option, `encoding_errors=ignore` to ignore the malformed data. While the option works as intended in `pd.read_csv` , but `wrangler.s3.read_csv`  doesn't seem to be honor this.

### How to Reproduce

Construct a file with non utf-8 characters:

```
import csv

def prepare_data(file_path):
    header = ["a", "b", "c", "d"]
    data = ["01AA1","15ÿ1515151515-63",511,511]

    with open(file_path, 'w') as f:
        writer = csv.writer(f)
        writer.writerow(header)
        writer.writerow(data)
``` 

Working with `read_csv`, pandas:
```
df = pd.read_csv(file_path, encoding='utf-8', encoding_errors='ignore')
```
Not working with Wrangler, throwing a `UnicodeDecodeError`:
```
df = wr.s3.read_csv(file_path, encoding='utf-8', encoding_errors='ignore') # throws UnicodeDecodeError
```

### Expected behavior

Should construct a `DataFrame` by ignoring malformed data.

```
      a                b    c    d
0  01AA1  151515151515-63  511  511
```

### OS

Windows 10

### Python version

3.9

### AWS SDK for pandas version

2.17

**Comment by cnfait** [comment]
There should be a fix in the next couple of days! as for a release that includes the fix, that is not decided yet.

**Comment by maciej-holderny** [comment]
Is there any update on this?

**Comment by cnfait** [comment]
I've proposed a fix today in this PR: #1723. It is now under review. You might be able to test it in your environment by installing the branch it is on:
```pip install git+https://github.com/aws/aws-sdk-pandas.git@read-csv-encoding-errors-param```

**Comment by maciej-holderny** [comment]
I can confirm it works properly right now - thanks. Just one note, there are also other values for encoding_errors which are not handled - example: "replace" - https://docs.python.org/3/library/codecs.html#error-handlers

**Comment by cnfait** [comment]
These are a bit different - I'll look into it although there may be more important changes coming soon to our `open_s3_object` code that would also handle these cases better.

**Comment by Sahana-Bhat-GOQ** [comment]
Thanks @cnfait for addressing the issue. Which release version can we expect to consume these changes?

**Comment by cnfait** [comment]
Hi @Sahana-Bhat-GOQ sorry for not coming back to you sooner. It will be released in version 2.18.0 which is not available yet but due this week.
